### PR TITLE
Datalog fixups

### PR DESCRIPTION
**Pull request by MingweiSamuel**
Been working on the website

### CODE REVIEW

File: hydroflow_datalog_core/src/lib.rs [hunk 1]
```diff
@@ -73,7 +73,13 @@ pub fn gen_hydroflow_graph(
             syn::LitInt::new(&format!("{}", my_merge_index), Span::call_site());
         let name = syn::Ident::new(&target.name, Span::call_site());
 
-        let input_pipeline: Pipeline = syn::parse_str(&hf_code.code).unwrap();
+        let input_pipeline: Pipeline = syn::parse_str(&hf_code.code).map_err(|err| {
+            vec![Diagnostic {
+                span: err.span(),
```
Comment (MingweiSamuel) [inline hydroflow_datalog_core/src/lib.rs:78]: Ok this span isn't actually right cause it's offset to the backticks

Comment (MingweiSamuel) [reply hydroflow_datalog_core/src/lib.rs:78]: Also it's not a real span anyway?

File: hydroflow_datalog_core/src/grammar.rs [hunk 2]
```diff
@@ -30,12 +32,17 @@ pub mod datalog {
     pub struct RustSnippet {
         #[rust_sitter::leaf(text = "`")]
         _start: (),
-        #[rust_sitter::leaf(pattern = r#"[^`]*"#, transform = |s| s.to_string())]
-        pub code: String,
+        pub code: Spanned<RustSnippetInner>,
```
Comment (shadaj) [inline hydroflow_datalog_core/src/grammar.rs:35]: released Rust Sitter 0.3.1 which should make it possible to go back to just `code: Spanned<String>`

**Review by shadaj — APPROVED** [review]
LGTM, I will also take a pass over this shortly to use spans on everything to have better error reporting.

**Comment by shadaj** [comment]
yea that compilefail has been flaky because which traits are available depends on unrelated features / dependencies

**Pull request merged**